src/main/java/org/reactome/addlinks/kegg/KEGGReferenceDatabaseGenerator.java

```diff
@@ -194,43 +200,48 @@ public static String generateDBNameFromReactomeSpecies(ReferenceObjectCache obje
 
 			if (keggSpeciesName != null)
 			{
-				targetDB = "KEGG Gene (" + keggSpeciesName + ")";
+				targetDB = KEGG_GENE + " (" + keggSpeciesName + ")";
 			}
 		}
 		return targetDB;
 	}
-	
+
 	/**
 	 * Generates a ReferenceDatabase name based on a KEGG species code. This will do a look-up in the cache for the species code to get the KEGG species name.
-	 * @param objectCache 
+	 * @param objectCache
 	 * @param keggSpeciesCode A KEGG species code.
 	 * @return "Kegg Gene ("+keggSpeciesName+")"
 	 */
 	public static String generateDBNameFromKeggSpeciesCode(ReferenceObjectCache objectCache, String keggSpeciesCode)
 	{
 		String targetDB = null;
-		
+
 		String keggSpeciesName = KEGGSpeciesCache.getSpeciesName(keggSpeciesCode);
-		String possibleDBName = "Kegg Gene ("+keggSpeciesName+")";
+		String possibleDBName = KEGG_GENE + " ("+keggSpeciesName+")";
 		if (objectCache.getRefDbNamesToIds().keySet().contains(possibleDBName))
 		{
 			targetDB = possibleDBName;
 		}
 		return targetDB;
 	}
-	
+
 	/**
 	 * To be used when you have a valid KEGG identifier but you can't figure out which Reactome species it should be for because there
-	 * is a different between the KEGG and Reactome species names. Use this method to create a new ReferenceDatabase object based on the KEGG 
+	 * is a different between the KEGG and Reactome species names. Use this method to create a new ReferenceDatabase object based on the KEGG
 	 * prefix and species name. Rebuilding caches is <em>STRONGLY</em> recommended after calling this method.
 	 * @param keggPrefix
 	 * @param keggSpeciesName
 	 * @return The name of the new ReferenceDatabase.
 	 */
 	public static synchronized String createReferenceDatabaseFromKEGGData(String keggPrefix, String keggSpeciesName, ReferenceObjectCache objectCache)
 	{
+		String newDBName = KEGG_GENE + " ("+keggSpeciesName + ")";
+		if (keggSpeciesName.contains("Mycobacterium tuberculosis H37Rv")) // if we start needing to handle more exceptional cases than just Mycobacterium tuberculosis H37Rv, we may need to rethink the general design...
```

Comment:
I wonder when other species in KEGG will also be subdivided into different strains

Reply:
Other species of single-cell organisms probably _are_. But so far, we have only had to deal with this one.